test/framework/resources/aws/services/eks.go

```diff
@@ -22,14 +22,24 @@ import (
 
 type EKS interface {
 	DescribeCluster(clusterName string) (*eks.DescribeClusterOutput, error)
-	CreateAddon(addon string, clusterName string) (*eks.CreateAddonOutput, error)
+	CreateAddon(createAddOnParams CreateAddOnParams) (*eks.CreateAddonOutput, error)
 	DeleteAddon(addon string, clusterName string) (*eks.DeleteAddonOutput, error)
+	DescribeAddonVersions(addon string, k8sVersion string) (*eks.DescribeAddonVersionsOutput, error)
+	DescribeAddon(addon string, clusterName string) (*eks.DescribeAddonOutput, error)
 }
 
 type defaultEKS struct {
 	eksiface.EKSAPI
 }
 
+type CreateAddOnParams struct {
```

Comment:
nit: may be rename to AddOnInput so that it can be extended for Describe, Delete as well if needed.  Also add a comment when why are we having this struct.